src/service/gif.js

```diff
@@ -1,9 +1,13 @@
 import api from './config';
 
-export const getGifFeatured = async () =>
+export const getGifFeatured = async (search) =>
   new Promise((resolve, reject) => {
     api
-      .get('/chat/gif/featured')
+      .get(
+        search?.trim() !== ''
+          ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}`
+          : '/chat/gif/featured'
+      )
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of `getGifFeatured` function is more flexible as it now supports searching for gifs. However, the current implementation could potentially lead to an injection attack since the search parameter is directly used in the API endpoint without proper sanitization or encoding. To prevent this, you should use `encodeURIComponent` function to encode the search string before appending it to the URL.

```diff
- ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}`
+ ? `/chat/gif/search?q=${encodeURIComponent(search)}`
```

Also, error handling seems to be missing for the promise. It's a good practice to handle potential errors that might occur during the execution of the promise. You can add a `.catch` block to handle any errors that might occur during the API call.

```diff
  api
    .get(
      search?.trim() !== ''
        ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}`
        : '/chat/gif/featured'
    )
    .then((res) => {
      resolve(res.data);
    })
+   .catch((error) => {
+     reject(error);
+   });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->